common/geth/multihoming_client.go

```diff
@@ -124,6 +137,7 @@ func (m *MultiHomingClient) HeaderByNumber(ctx context.Context, number *big.Int)
 func (m *MultiHomingClient) EstimateGas(ctx context.Context, msg ethereum.CallMsg) (uint64, error) {
 	var errLast error
 	for i := 0; i < m.NumRetries+1; i++ {
+		m.sleepBeforeRetry(i)
```

Comment:
nit: having this for loop with sleep copy-pasted everywhere suggests there is a nicer refactor potentially with a retry service that could be made. not critical though